src/metabase/driver/sql/query_processor.clj

```diff
@@ -274,6 +274,18 @@
 
 (defmethod ->honeysql [:sql :value] [driver [_ value]] (->honeysql driver value))
 
+(defmulti ^String expression-name->alias
+  "Return the String alias that should be used for `expression-name`, to follow the `AS` clause. This is to allow for
+  translating names that particular databases may not allow as identifiers for custom expressions (even when quoted).
+  Defaults to identity (i.e. returns `expression-name` unchanged)."
+  {:added "0.41.0" :arglists '([driver expression-name])}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
```

Comment:
I was actually considering adding something like this but more general, called `clause-alias`. Then we  can combine the existing `field->alias` method into it... 

https://github.com/metabase/metabase/pull/16656/files#diff-4fc60f4638cf98beb78946404682d4f857c629a070a01fc2d30112da04347b96R42-R46

A more general method might be better so we don't need to add `aggregation-reference->alias` or other stuff down the line

Reply:
So just to confirm, there is no chance that a driver might want different behavior for translating the name of an aggregation reference versus a field alias?